src/llvm/instruction_collection.cr

```diff
@@ -18,4 +20,12 @@ struct LLVM::InstructionCollection
   private def llvm_first
     LibLLVM.get_first_instruction @basic_block
   end
+
+  def each : Nil
+    inst = llvm_first
```

Comment:
You could inline `llvm_first` in here (and remove it), since it's not used anywhere else:

```suggestion
    inst = LibLLVM.get_first_instruction @basic_block
```